src/compiler/program.ts

```diff
@@ -4320,97 +4322,110 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
         }
     }
 
-    function getVersionForDeprecationDiagnostics(reportInvalidIgnoreDeprecations: boolean) {
-        const version = typeScriptVersion || versionMajorMinor;
+    function getIgnoreDeprecationsVersion(): Version {
         const ignoreDeprecations = options.ignoreDeprecations;
         if (ignoreDeprecations) {
-            if (ignoreDeprecations === DeprecationVersion.v5_0 && version.startsWith("5.")) {
-                return;
-            }
-            else if (reportInvalidIgnoreDeprecations) {
-                createOptionValueDiagnostic("ignoreDeprecations", Diagnostics.Invalid_value_for_ignoreDeprecations);
+            const parsed = Version.tryParse(ignoreDeprecations);

```

Comment:
We talked about this and it seems like the intent is that we are not going to deprecate _anything_ between TS 5.0 and TS 6.0, so this is overkill, and we can just write `=== "5.0"` and error.